marshmallow_utils/links.py

```diff
@@ -57,19 +59,48 @@ def resolve(self, context=None, config=None):
         """Resolves in-place all the tracked link dictionaries."""
         config = config or self.config
         assert config, "Links config is empty."
+
         context = context or {}
-        for ns, link_objects in self._links.items():
+        parameters = self._links
+
+        for ns, keyed_parameters_array in parameters.items():
             if ns not in config:
                 raise Exception(f'Unknown links namespace: {ns}')
-            for links in link_objects:
-                for k, v in links.items():
-                    links[k] = self.base_url(
-                        host=self.host,
-                        path=config[ns][k].expand(**context, **v),
```

Comment:
I think it's more the objects we stick in ``**v`` that has to be correct. URITemplate can correctly expand the same keyword:

```python
>>> from uritemplate import URITemplate
>>> from werkzeug.datastructures import MultiDict
>>> URITemplate("/records{?params*}").expand({'params': [('a','b'),('a','c'),]})
'/records?a=b&a=c'
```

or e.g.:

``` python
>>> params = {'type': ['a','b']}
>>> multi_params = MultiDict()
>>> for k, v in params.items():
...     multi_params.setlist(k, v)
>>> URITemplate("/records{?params*}").expand({'params': list(multi_params.items(multi=True))})
'/records?type=a&type=b'
```

Reply:
will implement a dict to associative array :+1: (don't think we'll need to bring werkzeug in)